packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -387,6 +411,20 @@ public void onConnectionSuspended(int cause) {
 
   @Override
   public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
+    if (requestCode == REQUEST_CODE_RESOLVE_ERROR) {
+      // Deal with result of `onConnectionFailed` error resolution.
+      resolvingError = false;
+      if (resultCode == Activity.RESULT_OK) {
+        // Make sure the app is not already connected or attempting to connect
+        if (!googleApiClient.isConnecting() && !googleApiClient.isConnected()) {
+          googleApiClient.connect();
+        }
+      } else if (pendingOperation != null && pendingOperation.method.equals(METHOD_INIT)) {
```

Comment:
This code pass can only be active during on "init" step (when the dart side told us t connect to the GoogleAPI) or when the activity is returning active after a connection had been established before. In the former case we need to tell the dart side that their requested action (init) failed. In the later case there is no pending operation (since the reconnect is implicit) and there is nobody listening on the native side to receive this error.